src/datatypes/src/json.rs

```diff
@@ -375,8 +375,8 @@ fn encode_json_value_with_context<'a>(
         }
         Json::Array(arr) => {
             let list_value = encode_json_array_with_context(arr, expected_type, context)?;
-            let data_type = list_value.datatype().clone();
-            Ok((Value::List(list_value), (*data_type).clone()))
+            let datatype = ConcreteDataType::List(ListType::new(list_value.datatype()));
+            Ok((Value::List(list_value), datatype))
```

Comment:
I found there are a lot of confusion in our code base about whether the datatype of list related api should return its inner type or list type itself.